internal/state/config/preferences.go

```diff
@@ -0,0 +1,116 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/pelletier/go-toml/v2"
+	"github.com/spf13/viper"
+)
+
+// Preferences are options that can be set in the config file, globally or per context
+type Preferences map[string]any
+
+func (p Preferences) Get(key string) (any, bool) {
+	configKey := strings.ReplaceAll(strings.ToLower(key), "-", "_")
```

Comment:
The `Get`, `Set` and `Unset` methods share a lot of common code related to the config code and path splitting. Is there some way to make them share the behaviour?